### PR TITLE
MTC-698: API to get market status (open/closed)

### PR DESCRIPTION
**Pull request by GSailen**

### CODE REVIEW

File: src/MarginTrading.SettingsService.Services/MarketDayOffService.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        {
+            var intersecting = compiledSchedule.Where(x => IsBetween(currentDateTime, x.Start, x.End));
+
+            return intersecting.OrderByDescending(x => x.Schedule.Rank)
+                       .Select(x => x.Schedule).FirstOrDefault()?.IsTradeEnabled ?? true;
+        }
+
+        private static bool IsBetween(DateTime currentDateTime, DateTime start, DateTime end)
+        {
+            return start <= currentDateTime && currentDateTime < end;
```
Comment (Waqas-Zamir) [inline src/MarginTrading.SettingsService.Services/MarketDayOffService.cs]: `currentDateTime <= end`

File: src/MarginTrading.SettingsService/Controllers/ScheduleSettingsController.cs [hunk 2]
```diff
@@ -153,6 +157,35 @@ public async Task<List<CompiledScheduleContract>> StateList([FromBody] string[]
...
+            var allMarkets = await _marketRepository.GetAsync();
+            if (marketIds == null || !marketIds.Any())
+            {
+                marketIds = allMarkets.Select(x => x.Id).ToArray();
+            }
+            else
+            {
+                foreach (var marketId in marketIds)
+                {
+                    if (allMarkets.Select(x => x.Id).ToHashSet().Contains(marketId))
```
Comment (Waqas-Zamir) [inline src/MarginTrading.SettingsService/Controllers/ScheduleSettingsController.cs]: please move `allMarkets.Select(x => x.Id).ToHashSet()` outside the loop

File: src/MarginTrading.SettingsService.Core/Domain/ScheduleConstraint.cs [hunk 3]
```diff
@@ -1,11 +1,39 @@
...
+    public class ScheduleConstraint : IEquatable<ScheduleConstraint>
     {
         public DateTime? Date { get; set; }
-        public DayOfWeek? DayOfWeek { get; set; }
+        public DayOfWeek? DayOfWeek { get; set; } 
         public TimeSpan Time { get; set; }
+        
+        public ScheduleConstraintType GetConstraintType()
+        {
+            if (Date == null && DayOfWeek == default)
```
Comment (Waqas-Zamir) [inline src/MarginTrading.SettingsService.Core/Domain/ScheduleConstraint.cs]: this is a little confusing, may be check both properties for null, default or .HasValue
like
`Date == null && DayOfWeek == null`
OR
`Date == default && DayOfWeek == default`

File: src/MarginTrading.SettingsService.Core/Constants.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+namespace MarginTrading.SettingsService.Core
+{
+    public static class Constants
+    {
+        public const string PlatformMarketId = "PlatformScheduleMarketId";
```
Comment (andrey-anokhin) [inline src/MarginTrading.SettingsService.Core/Constants.cs]: It should be a setting, not a constant. Like we did in the trading core.

File: src/MarginTrading.SettingsService/Controllers/ScheduleSettingsController.cs [hunk 5]
```diff
@@ -153,6 +157,35 @@ public async Task<List<CompiledScheduleContract>> StateList([FromBody] string[]
...
+                    {
+                        continue;
+                    }
+                        
+                    throw new ArgumentException($"Market {marketId} does not exist", nameof(marketIds));
```
Comment (andrey-anokhin) [inline src/MarginTrading.SettingsService/Controllers/ScheduleSettingsController.cs]: Not sure about the exception, we could just filter out not existing and return data only for existing ones. But not critical.

File: src/MarginTrading.SettingsService.Services/MarketDayOffService.cs (hunk 1 above)
Comment (GSailen) [reply src/MarginTrading.SettingsService.Services/MarketDayOffService.cs]: There is a convention about time intervals in calculations: we include starting point, and exclude ending.

File: src/MarginTrading.SettingsService/Controllers/ScheduleSettingsController.cs (hunk 5 above)
Comment (GSailen) [reply src/MarginTrading.SettingsService/Controllers/ScheduleSettingsController.cs]: Yep, I think it's not a big deal.

File: src/MarginTrading.SettingsService.Services/MarketDayOffService.cs (hunk 1 above)
Comment (Waqas-Zamir) [reply src/MarginTrading.SettingsService.Services/MarketDayOffService.cs]: okay

**Pull request merged**